packages/trpc/server/routers/viewer/workflows.tsx

```diff
@@ -907,6 +930,20 @@ export const workflowsRouter = router({
               }
             });
           }
+
+          for (const eventTypeId of activeOn) {
+            if (smsReminderNumberNeeded) {
+              await upsertSmsReminderFieldForBooking({
+                workflowId: id,
+                isSmsReminderNumberRequired: steps.some(
+                  (s) => s.action === WorkflowActions.SMS_ATTENDEE && s.numberRequired
+                ),
+                eventTypeId,
+              });
+            } else {
+              await removeSmsReminderFieldForBooking({ workflowId: id, eventTypeId });
+            }
+          }
```

Comment:
@CarinaWolli I just noticed that this is too much tracking that we have to do, due to this requirement of adding/updating/removing field in bookingFields.
One simplification I just noticed is that we can move this code at the very last of the fn(say line 1116). This would allow us to remove newly added code at 773, 732. This is possible because `activeOn` is the absolute truth we are going to save in DB.
Apart from that if something get's removed from activeOn, that's already handled at line 589.

Reply:
Good point, I made some changes to simplify it. @hariombalhara please take another look 🙏